docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,417 @@
+to the primary like so:
+
+```
+sudo crm resource move fs_res <primary_node_name>
+
+# remove the node constraint.
+sudo crm resource clear fs_res
+```
+
+### Kubernetes services
+
+We can now turn our attention to the Kubernetes services. Ensure that the k8s
+snap services no longer start automatically. Instead, they will be managed by a
+wrapper service.
+
+```
+for f in `sudo snap services k8s  | awk 'NR>1 {print $1}'`; do
+    echo "disabling snap.$f"
+    sudo systemctl disable "snap.$f";
+done
+```
+
+The next step is to define the wrapper service. Add the following to
+``/etc/systemd/system/2ha_k8s.service``. Note that the sample uses the
+``ubuntu`` user, feel free to use a different one as long as the prerequisites
+are met.
+
+```
+[Unit]
+Description=K8s service wrapper handling Dqlite recovery for 2-node HA setups.
+After=network.target pacemaker.service
+
+[Service]
+User=ubuntu
+Group=ubuntu
+Type=oneshot
+ExecStart=/bin/bash /snap/k8s/current/k8s/hack/2ha.sh start_service
+ExecStop=/bin/bash sudo snap stop k8s
+RemainAfterExit=true
+
+[Install]
+WantedBy=multi-user.target
+```
+
+```{note}
+The ``2ha.sh start_service`` command used by the service wrapper automatically
+detects the expected Dqlite role based on the DRBD state and takes the
+necessary steps to bootstrap the Dqlite state directories, synchronize with the
+peer node (if available) and recover the database.
+```
+
+We need the ``2ha_k8s`` service to be restarted once a DRBD failover occurs.
+For that, we are going to define a separate service that will be invoked by
+Pacemaker. Create a file called
+``/etc/systemd/system/2ha_k8s_failover.service`` containing the following:
+
+```
+[Unit]
+Description=Managed by Pacemaker, restarts 2ha_k8s on failover.
+After=network.target home-ubuntu-workspace.mount
+
+[Service]
+Type=oneshot
+ExecStart=systemctl restart 2ha_k8s
+RemainAfterExit=true
+```
+
+Reload the systemd configuration and set ``2ha_k8s`` to start automatically.
+Notice that ``2ha_k8s_failover`` must not be configured to start automatically,
+but instead is going to be managed through Pacemaker.
+
+```
+sudo systemctl enable 2ha_k8s
+sudo systemctl daemon-reload
+```
+
+Make sure that both nodes have been configured using the above steps before
+moving forward.
+
+We can now define a new Pacemaker resource that will invoke the
+``2ha_k8s_failover`` service when a DRBD failover occurs.
+
+```
+sudo crm configure <<EOF
+primitive ha_k8s_failover_service systemd:2ha_k8s_failover op start interval=0 timeout=120 op stop interval=0 timeout=30
+order failover_after_fs mandatory: fs_res:start ha_k8s_failover_service:start
+colocation fs_failover_colo INFINITY: fs_res ha_k8s_failover_service
+commit
+show
+quit
+EOF
+```
+
+The setup is ready, start the HA k8s service on both nodes:
+
+```
+sudo systemctl start 2ha_k8s
+```
+
+## Troubleshooting
+
+### Dqlite recovery failing because of unexpected data segments
```

Comment:
I believe we need to capitalize more "important" words in our headers here and in other places.

Reply:
We could, although the other docs don't seem to do this:

* https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/external-datastore/

* https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/backup-restore/

Furthermore, it might affect readability.